datastore/elastic_handler.go

```diff
@@ -0,0 +1,170 @@
+package datastore
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"github.com/daniyakubov/book_service_n/config"
+	"github.com/daniyakubov/book_service_n/consts"
+	"github.com/daniyakubov/book_service_n/consts/elastic_fields"
+	"github.com/daniyakubov/book_service_n/models"
+	errors "github.com/fiverr/go_errors"
+	"github.com/olivere/elastic/v7"
+	"strconv"
+	"strings"
+)
+
+var _ BookStorer = &ElasticHandler{}
+
+type ElasticHandler struct {
+	Client *elastic.Client
+}
+
+func NewElasticHandler(client *elastic.Client) ElasticHandler {
+	return ElasticHandler{
+		client,
+	}
+}
+
+func (e *ElasticHandler) UpdateBook(ctx context.Context, title string, id string) (err error) {
+	_, err = e.Client.Update().
+		Index(consts.Index).
+		Id(id).
+		Doc(map[string]interface{}{elastic_fields.Title: title}).
+		Do(ctx)
+	if err != nil {
+		return errors.Wrap(err, fmt.Sprintf("couldn't update book with title %s and id %s", title, id))
+	}
+
+	return nil
+}
+
+func (e *ElasticHandler) InsertBook(ctx context.Context, body []byte) (string, error) {
+	res, err := e.Client.Index().
+		Index(consts.Index).
+		BodyString(string(body)).
+		Do(ctx)
+	if err != nil {
+		return "", errors.Wrap(err, fmt.Sprintf("couldn't update book with body: %s", string(body)))
+	}
+
+	return res.Id, nil
+}
+func (e *ElasticHandler) GetBook(ctx context.Context, id string) (src *models.Book, err error) {
+	get, err := e.Client.Get().
+		Index(consts.Index).
+		Id(id).
+		Do(ctx)
+	if err != nil {
+		return nil, errors.Wrap(err, fmt.Sprintf("couldn't get book form server with id: %s", id))
+	}
+
+	var book models.Book
+	if err = json.Unmarshal(get.Source, &book); err != nil {
+		return nil, errors.Wrap(err, fmt.Sprintf("couldn't unmarshal result of book with id: %s, in getBoook function ", id))
+	}
+
+	return &book, nil
+}
+
+func (e *ElasticHandler) DeleteBook(ctx context.Context, id string) error {
+	_, err := e.Client.Delete().
+		Index(consts.Index).
+		Id(id).
+		Do(ctx)
+	if err != nil {
+		return errors.Wrap(err, fmt.Sprintf("couldn't delete book with id: %s", id))
+	}
+
+	return nil
+}
+
+func buildSearchQuery(fields map[string]string) (s *elastic.BoolQuery, err error) {
+	var from, to int
```

Comment:
I think it is better to declare `q` at the begging of the function and start with the other fields, and at the end handle `price_range` param - but not critical